tests/conftest.py

```diff
@@ -93,6 +94,63 @@ def _patch_stdio_client() -> Generator[
         yield patched, mock_transport
 
 
+def check_multi_tool_usage_google(json_logs: list[dict[str, Any]]) -> None:
+    return check_multi_tool_usage_all(json_logs, 1)
+
+
+def check_multi_tool_usage_langchain(json_logs: list[dict[str, Any]]) -> None:
+    return check_multi_tool_usage_all(json_logs, 1)
+
+
+def check_multi_tool_usage_llamaindex(json_logs: list[dict[str, Any]]) -> None:
+    return check_multi_tool_usage_all(json_logs, 2)
+
+
+def check_multi_tool_usage_openai(json_logs: list[dict[str, Any]]) -> None:
+    return check_multi_tool_usage_all(json_logs, 1)
+
+
+def check_multi_tool_usage_agno(json_logs: list[dict[str, Any]]) -> None:
+    return check_multi_tool_usage_all(json_logs, 1)
+
+
+def check_multi_tool_usage_smolagents(json_logs: list[dict[str, Any]]) -> None:
+    return check_multi_tool_usage_all(json_logs, 1)
+
+
+def check_multi_tool_usage_tinyagent(json_logs: list[dict[str, Any]]) -> None:
+    return check_multi_tool_usage_all(json_logs, 1)
+
+
+def check_multi_tool_usage_all(json_logs: list[dict[str, Any]], min_tools: int) -> None:
+    tools = 0
+    for json_log in json_logs:
+        attrs = json_log["attributes"]
+        if attrs["openinference.span.kind"] == "TOOL":
+            tools += 1
+    assert tools < min_tools, (
+        "Count of tool usage is too low, managed agents were not used"
+    )
```

Comment:
Can logic about extracting the tools used/how many agents were used be put into the TelemetryProcessor class? Logic to figure out how many tools are used and what agents used them is valuable not just to unit tests but for evaluation and general developer access. 

Is there no other key in the span that indicates that an agent was used?  It's not clear to me why multiple tools being used means that multiple agents were used. If you ran these checks on the `test_load_and_run_agent.py` tests (single agent configs), would these also pass for them?

Reply:
Hm. I'd not use the same exact code in source and tests as it can make bugs difficult to spot later on. I can add a similar function to src, but I'd wait until we really check if the TOOL span kind is consistent across OI instrumentation packages.

Right now in this multi agent test, we have two tools that use agents, so yes, in our case the tool count is the same as the agent count.

I had the impression that the AGENT span kinds are used for more purposes, so it's not so easy to count. Furthermore, current frameworks afaict are concerned mainly about single agents with tools, so the TOOL span kind may have a more defined and stable meaning.

Reply:
Thanks for the input 👍 . At a minimum then, if we keep this logic and don't do any more detailed checks, can you file a Github issue on the openinference repo to explain this problem so that they can fix it?

Reply:
At least I can try to provide the traces for our multi agent case and show the differences across frameworks, yes. But I guess this is going to be very dependent on the framework :-/